### PR TITLE
KFSPTS-19988 adding 1099 form type to labor object code to 1099 box mapping

### PR DESCRIPTION
**Pull request by jhulslander**
Please note there is a related non-prod-sql PR that should be merged with this.

### CODE REVIEW

File: src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml:11]: Please see my note on the related SQL PR. If the "FORM_TYPE" does become part of the primary key, then either this user story or the follow-up user story should update the OJB metadata to configure it as such.

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:9]: Can this import be removed?

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:25]: In this test method and the one below it, I would suggest including an assertion that also checks the formCode field.

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:46]: You may also want a test case for a search value that is non-blank but invalid.

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:17]: I'm noticing a lot of tab characters in the new and modified code sections of this PR's files. Can these tabs be converted into spaces instead?

**Review by cah292 — CHANGES_REQUESTED** [review]
I had a few suggestions for improvement, and one note about a change that may need to be added in the follow-up user story. Other than that, the changes look good.

File: src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml:11]: I have an open question to the functionals on this

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:9]: yes it should

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:25]: OK

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:46]: ok

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:17]: ok

**Comment by jhulslander** [comment]
I've made the changes suggested, there is an open question about object code / payment reason code mapping to more than one form type.  We should get an answer to that before merging.

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:20]: Our cu-kfs Java code typically uses 4 spaces per indentation level. Can you please adjust the indentation of this PR's code for consistency with that? Thanks!

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:50]: I would suggest removing the word "Empty" from the method name, to more accurately describe the search input that is being tested.

**Review by cah292 — CHANGES_REQUESTED** [review]
I found a couple minor naming and indentation issues, but otherwise the changes look good. I will continue to wait for the functional feedback on whether any primary key adjustments are needed.

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:20]: ok

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:50]: ok

**Comment by jhulslander** [comment]
OK, I spoke with Cathy Salino for right now, we only want an object code and payment reason code mapped to only one box on one form
So I don't think the primary key needs to be adjusted.  This is ready for review and can be merged when you think it is ready

File: src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml:11]: Thanks for confirming this with the functionals. Based on the feedback you provided, I'm fine with the form type as-is.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass, and the related SQL PR is also ready. Merging.

**Pull request merged**
